orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -95,35 +95,45 @@ Status CreateTensorProtosFromOrtValues(
   return Status::OK();
 }
 
-PathString GetTensorProtoFilePath(const PathString& checkpoint_directory, const std::string& filename_prefix) {
-  return ConcatPathComponent<PathChar>(checkpoint_directory, ORT_TSTR(filename_prefix + k_name_seperator) + k_tensor_proto_file_name);
+PathString GetTensorProtoFilePath(const PathString& checkpoint_directory, const PathString& filename_prefix) {
+  std::basic_ostringstream<PathChar> oss;
```

Comment:
Why do we use `ostringstream` instead of `ConcatPathComponent`?

Reply:
one is to compose a file name, the other one is to concat a path just like ops.path.join. They are different, right?

Reply:
oh sorry, I was not clear. What I meant was is `std::basic_ostringstream<PathChar>` necessary for composing the path?
I was wondering if simply using string concat would not work (str1+str2)?
I think this change is fine. Just wanted to understand the reason behind using ostringstream instead of string concat.